### DISCUSSION TITLE
How to access parent row data in subRow when using `useExpanded`?

### DISCUSSION THREAD

**Discussion by vinnymac** [TanStack/table#2044] [General]
I was able to get access to the first row of data by doing the following.

```ts
const columns = useMemo(() => ({
  Cell({
    row: { id },
    rowsById
  }) {
    const { original } = rowsById[id[0]];
    return <span>{original.firstName}</span>;
  }
}));
```

This technique could be used to access any of the relevant rows. Was this the intended way of accessing the parent rows from a subRow? Is there a better way that I am not seeing?

Thank you for reading

**Comment by victmo** [comment]
This is how I do it:
```js
const parentRowId = row.id.split('.').slice(0, -1).join('.'); // string
const parentRow = instance.rowsById[parentRowId]; // Row || undefined;
```

**Reply by YuriyDyachkov** [reply]
> This is how I do it:
> 
> ```js
> const parentRowId = row.id.split('.').slice(0, -1).join('.'); // string
> const parentRow = instance.rowsById[parentRowId]; // Row || undefined;
> ```

Hello. Whats is this "instance"?

**Reply by victmo** [reply]
`instance` is what gets returned from `useTable`. See https://react-table.tanstack.com/docs/api/overview#hook-usage

**Comment by T04435** [comment]
I'm getting access to it from the columns in the table.

For that I use: `row.getParentRow()?.original` 


```
export const columns: ColumnDef<T>[] = [
  {  
    accessorKey: "key",
    header: "header",
    cell: ({ row, column, cell }) => {
      const data = row.original // parent || child
      const parentRow = row.getParentRow()?.original // parent or undefined if parent
      return (...)
    }
  }
  // ... other cols
] 
```

**Reply by MDziurakh** [reply]
@T04435, definitely should be on top!
